test/jdk/java/awt/event/ComponentEvent/TextAreaTextEventTest.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2007, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.TextArea;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+
+/*
+ * @test
+ * @key headful
+ * @bug 8296632
+ * @summary Verify the content changes of a TextArea via TextListener.
+ * @run main TextAreaTextEventTest
+ */
+public class TextAreaTextEventTest {
+
+    private static Frame frame;
+    private volatile static TextArea textArea;
+    private volatile static boolean textChanged = false;
+    private volatile static Point textAreaAt;
+    private volatile static Dimension textAreaSize;
+    private static Robot robot = null;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            EventQueue.invokeAndWait(TextAreaTextEventTest::initializeGUI);
+
+            robot = new Robot();
+            robot.setAutoDelay(100);
+            robot.setAutoWaitForIdle(true);
+
+            robot.waitForIdle();
+
+            EventQueue.invokeAndWait(() -> {
+                textAreaAt = textArea.getLocationOnScreen();
+                textAreaSize = textArea.getSize();
+            });
+
+            robot.mouseMove(textAreaAt.x + textAreaSize.width / 2,
+                textAreaAt.y + textAreaSize.height / 2);
+
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+            typeKey(KeyEvent.VK_T);
+
```

Comment:
In theory the autoWaitForIdle should ensure that events are processed. We'll see.
If you find otherwise then here (and after typing the Enter key) it might be safer to do an additional robot.delay(500);
You want to give every chance to process these events.

Reply:
Tested it locally on Win10. Works as expected. I agree @prrace, adding additional delay would make sure that all events are processed.

Reply:
Without explicit `Robot.waitForIdle()` calls, the test looks as if something's missing. Adding it explicitly before checking a condition gives more confidence, the test has waited for the events to be processed.

Reply:
Called Robot.waitForIdle() explicitly to make sure that all events are processed before checking a condition.

Reply:
@aivanov-jdk A quick clarification - Since `robot.setAutoWaitForIdle(true);` is set to true would it be required to add `waitForIdle()` calls explicitly too?

https://github.com/openjdk/jdk/blob/9c432a09dbf1824d793cc1a49b533e57f2316b62/src/java.desktop/share/classes/java/awt/Robot.java#L606